quantus/functions/explanation_func.py

```diff
@@ -352,42 +387,37 @@ def f_reduce_axes(a):
 
     explanation: torch.Tensor = torch.zeros_like(inputs)
 
-    if method == "GradientShap".lower():
+    if method in ["GradientShap", "IntegratedGradients", "DeepLift", "DeepLiftShap"]:
+        attr_func = eval(method)
         explanation = f_reduce_axes(
-            GradientShap(model).attribute(
+            attr_func(model).attribute(
                 inputs=inputs,
                 target=targets,
                 baselines=kwargs.get("baseline", torch.zeros_like(inputs)),
             )
         )
 
-    elif method == "IntegratedGradients".lower():
+    elif method in [
+        "InputXGradient",
+        "Saliency",
+        "FeatureAblation",
+        "Deconvolution",
+        "FeaturePermutation",
+        "Lime",
+        "KernelShap",
+        "LRP",
+    ]:
+        attr_func = eval(method)
         explanation = f_reduce_axes(
-            IntegratedGradients(model).attribute(
-                inputs=inputs,
-                target=targets,
-                baselines=kwargs.get("baseline", torch.zeros_like(inputs)),
-                n_steps=10,
-                method="riemann_trapezoid",
-            )
-        )
-
-    elif method == "InputXGradient".lower():
-        explanation = f_reduce_axes(
-            InputXGradient(model).attribute(inputs=inputs, target=targets)
+            attr_func(model).attribute(inputs=inputs, target=targets)
```

Comment:
What if the user has kwargs for the explanation method? Where does it go? Not in attr_func?

Reply:
@dilyabareeva check this when you can :)